crates/diesel_models/src/query/authentication.rs

```diff
@@ -0,0 +1,76 @@
+use diesel::{associations::HasTable, BoolExpressionMethods, ExpressionMethods};
+use router_env::{instrument, tracing};
+
+use super::generics;
+use crate::{
+    authentication::{
+        Authentication, AuthenticationNew, AuthenticationUpdate, AuthenticationUpdateInternal,
+    },
+    errors,
+    schema::authentication::dsl,
+    PgPooledConn, StorageResult,
+};
+
+impl AuthenticationNew {
+    #[instrument(skip(conn))]
+    pub async fn insert(self, conn: &PgPooledConn) -> StorageResult<Authentication> {
+        generics::generic_insert(conn, self).await
+    }
+}
+
+impl Authentication {
+    #[instrument(skip(conn))]
```

Comment:
Please remove `instrument` from methods in the `diesel_models::query` module and add it to the DB interface methods in the `router::db` module instead.

For more information, check: #3838